utils/swift_build_support/swift_build_support/products/llvm.py

```diff
@@ -216,20 +216,11 @@ def build(self, host_target):
                 # space/time efficient than -g on that platform.
                 llvm_cmake_options.define('LLVM_USE_SPLIT_DWARF:BOOL', 'YES')
 
-        build_targets = ['all']
-
-        if self.args.llvm_ninja_targets_for_cross_compile_hosts and \
-           self.is_cross_compile_target(host_target):
-            build_targets = (self.args.llvm_ninja_targets_for_cross_compile_hosts)
-        elif self.args.llvm_ninja_targets:
-            build_targets = (self.args.llvm_ninja_targets)
-
-        # indicating we don't want to build LLVM should
-        # override any custom ninja target we specified
         if not self.args._build_llvm:
-            build_targets = ['clean']
```

Comment:
I’m not sure why we used to add the `clean` target but it looks like we lost it. Is that intentional?

Reply:
Neither am I, but I doubt it makes a difference.

> Is that intentional?

Yes, I changed the logic slightly to not run any build commands for LLVM when `--build-llvm=0` (the "actually don’t build LLVM" option), so it no longer matters what `build_targets` is set to in this case. This behavior is consistent with how we handle other CMake products, and it allows the new test to not just succeed (a build command that was assembled for the Xcode generator will fail against a Ninja build directory due to [incompatible adjustment to the options](https://github.com/apple/swift/blob/7eac19998490f2912937e1ef271b4c43f8c12164/utils/swift_build_support/swift_build_support/products/cmake_product.py#L76-L87), but also run correctly (we don’t want to run any build commands against the symlinked LLVM build directory).

Reply:
Sounds reasonable to me 👍🏽